src/coreclr/debug/createdump/threadinfo.cpp

```diff
@@ -272,6 +192,92 @@ ThreadInfo::UnwindThread(IXCLRDataProcess* pClrDataProcess)
+    GetFrameLocation(pContext, &ip, &sp);
+
+    uint64_t moduleAddress = 0;
+    mdMethodDef token = 0;
+    uint32_t nativeOffset = 0;
+    uint32_t ilOffset = 0;
+
+    ReleaseHolder<IXCLRDataFrame> pFrame;
+    if (SUCCEEDED(pStackwalk->GetFrame(&pFrame)))
+    {
+        CLRDataSimpleFrameType simpleType;
+        CLRDataDetailedFrameType detailedType;
+        pFrame->GetFrameType(&simpleType, &detailedType);
+
+        if ((simpleType & (CLRDATA_SIMPFRAME_MANAGED_METHOD | CLRDATA_SIMPFRAME_RUNTIME_MANAGED_CODE)) != 0)
+        {
+            ReleaseHolder<IXCLRDataMethodInstance> pMethod;
+            if (SUCCEEDED(pFrame->GetMethodInstance(&pMethod)))
+            {
+                ReleaseHolder<IXCLRDataModule> pModule;
+                if (SUCCEEDED(pMethod->GetTokenAndScope(&token, &pModule)))
+                {
+                    DacpGetModuleData moduleData;
+                    if (SUCCEEDED(moduleData.Request(pModule)))
+                    {
+                        moduleAddress = moduleData.LoadedPEAddress;
+                    }
+                    else
+                    {
+                        TRACE("Unwind: DacpGetModuleData.Request sp %p ip %p FAILED\n", (void*)sp, (void*)ip);
+                    }
+                }
+                else
+                {
+                    TRACE("Unwind: GetTokenAndScope sp %p ip %p FAILED\n", (void*)sp, (void*)ip);
+                }
+                if (FAILED(pMethod->GetILOffsetsByAddress(ip, 1, NULL, &ilOffset)))
+                {
+                    TRACE("Unwind: GetILOffsetsByAddress sp %p ip %p FAILED\n", (void*)sp, (void*)ip);
+                }
+                CLRDATA_ADDRESS startAddress;
+                if (SUCCEEDED(pMethod->GetRepresentativeEntryAddress(&startAddress)))
+                {
+                    nativeOffset = ip - startAddress;
+                }
+                else
+                {
+                    TRACE("Unwind: GetRepresentativeEntryAddress sp %p ip %p FAILED\n", (void*)sp, (void*)ip);
+                }
+            }
+            else
+            {
+                TRACE("Unwind: GetMethodInstance sp %p ip %p FAILED\n", (void*)sp, (void*)ip);
+            }
+        }
+        else
+        {
+            TRACE("Unwind: simpleType %08x detailedType %08x\n", simpleType, detailedType);
+        }
+    }
+
+    // Add managed stack frame for the crash info notes
+    StackFrame frame(moduleAddress, ip, sp, nativeOffset, token, ilOffset);
+    AddStackFrame(frame);
+}
+
+void
+ThreadInfo::AddStackFrame(const StackFrame& frame)
+{
+    const auto& found = m_frames.find(frame);
+    if (found == m_frames.end())
+    {
+        TRACE("Unwind: sp %p ip %p off %08x mod %p%c\n",
+            (void*)frame.StackPointer(),
+            (void*)frame.ReturnAddress(),
+            frame.NativeOffset(),
+            (void*)frame.ModuleAddress(),
+            frame.IsManaged() ? '*' : ' ');
+        m_frames.insert(frame);
```

Comment:
This call you can do, and you don't need a find. It returns a pair, where the second element is a bool that tells you if insertion was necessary.